### PR TITLE
Feat playlist

### LINKED ISSUE #13: feat "Playlist"
This issue focuses on adding videos to the playlist and related functionalities.

### PR DESCRIPTION
**Pull request by vaibhav-mahalle**
- Added playlists functionality
- Added filter by categories
- Made the styling changes.
- Added favicon
- Added HomeBanner

### CODE REVIEW

File: src/Pages/History/index.js [hunk 1]
```diff
@@ -13,17 +13,20 @@ export const History = () => {
...
-        <div className="common-videolisting-container">
-          {history.map((item) => {
-            return <VideoCon item={item} functionType="deleteHistory"/>;
-          })}
-        </div>
+        <>
+          <h2 className="common-page-heading p-1"> History --- {history.length}</h2>
+          <div className="common-videolisting-container">
+            {history.map((item) => {
+              return <VideoCon item={item} functionType="deleteHistory" />;
```
Comment (srejitk) [inline src/Pages/History/index.js:26]: Provide a key here

File: src/Pages/Home/Home.css [hunk 2]
```diff
@@ -1,4 +1,11 @@
...
-}
\ No newline at end of file
+}
+
+/* .align-content-grid{
+    display: grid;
+    border: 5px solid red ;
+    grid-template-columns: repeat(auto-fit, (300px,1fr));
+    gap: 1rem 1rem;
+} */
```
Comment (srejitk) [inline src/Pages/Home/Home.css:11]: Remove the commented code if not needed

File: src/Pages/Playlist/Playlist.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          <div className="msg-container txt-xlg">
+            You haven't added anything
+            <Link to="/explore" className="txt-white-color p-l-1">
+              watch now
+            </Link>
+          </div>
+        </div>
+      ) : (
+        <>
+          <h2 className="common-page-heading p-1"> Playlists --- {playlists.length}</h2>
```
Comment (srejitk) [inline src/Pages/Playlist/Playlist.js:23]: Add optional chaining here to prevent undefined error

File: src/Pages/Home/Home.css (hunk 2 above)
Comment (sadathshariff) [inline src/Pages/Home/Home.css:11]: Remove the code here

File: src/Pages/Playlist/Playlist.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+          <div className="common-videolisting-container">
+            {playlists.map(({ _id, title}) => (
```
Comment (srejitk) [inline src/Pages/Playlist/Playlist.js:25]: Here too

File: src/Pages/Playlist/SinglePlaylist.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    try {
+      const res = await axios.get(`/api/user/playlists/${playlistId}`, {
+        headers: {
+          authorization: isLoggedIn,
+        },
+      });
+
+      if (res.status === 200) {
+        setSinglePlaylistVideos(res.data.playlist);
+        console.log("playlist", res.data.playlist);
```
Comment (sadathshariff) [inline src/Pages/Playlist/SinglePlaylist.js:27]: Remove console.log

File: src/components/CategoryBox/CategoryBox.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+import { useVideo } from "../../context";
+
+
+export const CategoryBox = () => {
+
+    const {loadingData, responseData, errorData} = useAxios(`/api/categories`);
+    const {videoDispatch} = useVideo();
+
+    const categories = responseData.categories || [];
+    console.log(categories)
```
Comment (sadathshariff) [inline src/components/CategoryBox/CategoryBox.jsx:14]: Here as well

**Review by sadathshariff — APPROVED** [review]
Everything looks good, suggested few points look into it

File: src/Pages/Error/index.js [hunk 7]
```diff
@@ -1,9 +1,16 @@
-import React from 'react'
+import { Link } from "react-router-dom";
+import React from "react";
+Link
```
Comment (rahulcs754) [inline src/Pages/Error/index.js:3]: i think here you right by mistake "Link"

File: src/components/Modal/Modal.jsx [hunk 8]
```diff
@@ -1,10 +1,110 @@
...
+  useEffect(() => {
+    getAllPlaylist(isLoggedIn, videoDispatch);
+  }, [playlists]);
 
-const Modal =() =>{
   return (
-    <div>Modal</div>
-  )
-}
+    <div className="modal-box">
```
Comment (srejitk) [inline src/components/Modal/Modal.jsx:48]: Consider using React Portal here for better support. Going with the current approach can hinder where your Modal appears

File: src/Pages/Home/Home.css (hunk 2 above)
Comment (rahulcs754) [inline src/Pages/Home/Home.css:11]: remove this comment code

**Review by srejitk — APPROVED** [review]
Code looks good. Just suggested a few minor changes. Keep at it! Good Work!

File: src/context/Video/playlist.js [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+    );
+    if (res.status === 200) {
+      videoDispatch({
+        type: "REMOVE_VIDEO_FROM_PLAYLIST",
+        payload: res.data.playlist,
+      });
+      ToastMsg("Video removed from Playlist", "success");
+    }
+  } catch (error) {
+    console.log(error);
```
Comment (rahulcs754) [inline src/context/Video/playlist.js:129]: i think you should remove console log from your code

**Review by rahulcs754 — APPROVED** [review]
Just few suggestions

- remove console log before push
- you should  remove comment code from your project

**Pull request merged**